lib/manager/terraform/lockfile/hash.ts

```diff
@@ -31,7 +32,7 @@ export class TerraformProviderHash {
 
       // add double space, the filename and a new line char
       rootHash.update('  ');
-      const fileName = file.replace(/^.*[\\/]/, ''); // TODO #12070
+      const fileName = file.replace(regEx(/^.*[\\/]/), '');
```

Comment:
i think we should use a `upath` function to extract filename instead of regex.
```suggestion
      const fileName = upath.basename(file);
```

Reply:
Why do you resolved this without comment?

Reply:
Sorry for that I thought I committed this change.
Will it be okay to apply this change in #12071 where all similar changes will be handled?